src/api/app/controllers/source_package_meta_controller.rb

```diff
@@ -30,10 +30,9 @@ def show
     end
 
     # Let the backend answer for deleted or remote packages. For specific revisions or the blame view. Or if the meta parameter is used.
-    if params.key?(:deleted) || params.key?(:meta) || params.key?(:rev) || params.key?(:view) || pack.nil?
-      path = request.path_info
-      path += build_query_from_hash(params, %i[deleted meta rev view])
-      pass_to_backend(path)
+    meta_params = params.slice(:deleted, :meta, :rev, :view).permit!.to_h
+    if meta_params.any? || pack.nil?
```

Comment:
:thought_balloon:  This `pack` is not needed below, right? I mean, it does not make part of the condition body. Then you can just check if it's not nil and return before doing the `slice` and the rest?